test/test_examples.py

```diff
@@ -442,6 +442,102 @@ def test_cuse(output_checker):
     finally:
         mount_process.terminate()
 
+def test_746(tmpdir, output_checker):
+    # ###
+    # ### test case for Issue #746
+    # ###
+    #
+    # If RELEASE and UNLINK opcodes are sent back to back, and fuse_fs_release()
+    # and fuse_fs_rename() are slow to execute, UNLINK will run while RELEASE is
+    # still executing. UNLINK will try to rename the file and, while the rename
+    # is happening, the RELEASE will finish executing. As a result, RELEASE will
+    # not detect in time that UNLINK has happened, and UNLINK will not detect in
+    # time that RELEASE has happened.
+    #
+    #
+    # NOTE: This is triggered only when nullpath_ok is set.
+    #
+    # If it is NOT SET then get_path_nullok() called by fuse_lib_release() will
+    # call get_path_common() and lock the path, and then the fuse_lib_unlink()
+    # will wait for the path to be unlocked before executing and thus synchronise
+    # with fuse_lib_release().
+    #
+    # If it is SET then get_path_nullok() will just set the path to null and
+    # return without locking anything and thus allowing fuse_lib_unlink() to
+    # eventually execute unimpeded while fuse_lib_release() is still running.
+    #
+    #
+    # This test relies on an external library (test746.so) that is LD_PRELOADed
+    # to intercept the libc's close() and rename() calls and add a show delay
+    # before forwarding the calls to libc.
+    #
+    # The delay can be disabled at call-time by setting the TEST746_DELAY_DISABLE
+    # environment variable. e.g:
+    # TEST746_DELAY_DISABLE=1 python3 -m pytest test/test_examples.py::test_746
+
+    # set the path for the library to preload
+    preloadLib = base_cmdline + \
+        [ pjoin(basename, "test", "test746.so") ]
+    preloadLib = preloadLib[0]
+    #print("preloadLib: ", preloadLib)
+
+    fuseMNT = str(tmpdir)
+    #print("fuseMNT: ", fuseMNT)
+
+    # set the FUSE binary path and command
+    fuseCMD = base_cmdline + \
+        [ pjoin(basename, 'example', 'passthrough_fh'),
+        "-f", fuseMNT]
+    #print("fuseCMD: ", str(fuseCMD))
+
+    # start FUSE with our preload library
+    os.environ["LD_PRELOAD"] = preloadLib
+    fuseProcess = subprocess.Popen(fuseCMD,
+                                   stdout=output_checker.fd,
+                                   stderr=output_checker.fd)
+    os.environ["LD_PRELOAD"] = ""
+
+    leftoverFilesCount = -1
+    try:
+        wait_for_mount(fuseProcess, fuseMNT)
+
+        # use TemporaryDirectory so that it gets cleaned up automatically
```

Comment:
This comment just repeats what the code already says, please remove.

Reply:
This is a deliberate comment so that `TemporaryDirectory()` is not replaced in the future with `mkdtemp()`, which will need the user to handle deleting the directory and any leftover contents.

Reply:
Changes included in next patch.